src/platform/storage_winuser.c

```diff
@@ -150,30 +154,53 @@ CxPlatStorageOpen(
 
     REGSAM DesiredAccess = KEY_READ | KEY_NOTIFY;
 
-    if (Flags & CXPLAT_STORAGE_OPEN_FLAG_WRITABLE) {
+    if (Flags & CXPLAT_STORAGE_OPEN_FLAG_WRITEABLE) {
         DesiredAccess |= KEY_WRITE;
     }
 
     if (Flags & CXPLAT_STORAGE_OPEN_FLAG_DELETEABLE) {
         DesiredAccess |= DELETE;
     }
 
-#pragma prefast(suppress:6001, "SAL can't track FullKeyName")
-    Status =
-        HRESULT_FROM_WIN32(
-        RegOpenKeyExA(
-            HKEY_LOCAL_MACHINE,
-            FullKeyName,
-            0,
-            DesiredAccess,
-            &Storage->RegKey));
-    if (QUIC_FAILED(Status)) {
-        QuicTraceEvent(
-            LibraryErrorStatus,
-            "[ lib] ERROR, %u, %s.",
-            Status,
-            "RegOpenKeyExA failed");
-        goto Exit;
+    if (Flags & CXPLAT_STORAGE_OPEN_FLAG_CREATE) {
+        Status =
+            HRESULT_FROM_WIN32(
+            RegCreateKeyExA(
+                HKEY_LOCAL_MACHINE,
+                FullKeyName,
+                0,
+                NULL,
+                REG_OPTION_NON_VOLATILE,
+                DesiredAccess,
+                NULL,
+                &Storage->RegKey,
+                NULL));
+        if (QUIC_FAILED(Status)) {
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "[ lib] ERROR, %u, %s.",
+                Status,
+                "RegCreateKeyExA failed");
+            goto Exit;
+        }
+    } else {
+        #pragma prefast(suppress:6001, "SAL can't track FullKeyName")
```

Comment:
question: Did you check if this is still needed? It sounds strange it would be but the "create" path does not need it (+ I don't see why checkers would have troubles with it)

Reply:
let's try a CI run without it and see what happens